zerver/tests/test_users.py

```diff
@@ -826,7 +826,7 @@ def test_create_user_with_multiple_streams(self) -> None:
 
         prereg_user = PreregistrationUser.objects.get(email="fred@zulip.com")
 
-        with self.assert_database_query_count(94):
+        with self.assert_database_query_count(95):
```

Comment:
Why does this add an extra database query? This commit seems like it would not do any extra queries unless it's for `realm.can_access_all_users_group.name == SystemGroups.EVERYONE`?

Reply:
Indeed, changing https://github.com/zulip/zulip/pull/27544#discussion_r1401561350 to move the "are there any guests" check up caused this query to no longer be required.